### PR TITLE
Branch coverage issue with several awaits

### LINKED ISSUE #1176: Unexpected branch coverage with using statement and several awaits
I've encountered this with stable coverlet.collector as well as nightly 3.0.4-preview.31.g4902c245c8. I'm using .NET SDK 5.0.300 and running `dotnet test --collect:"XPlat Code Coverage"`.

This code has 3 out of 6 blocks covered despite having no branches apart from awaits. Removing or reordering any of the statements results in 100% coverage and cyclomatic complexity of 1 instead of 6. I would expect this code to also have cyclomatic complexity of 1 and to be fully covered.

```csharp
using System.IO;
using System.Threading.Tasks;

namespace Lib
{
    public class Foo
    {
        public static async Task Bar()
        {
            await Task.CompletedTask;
            using var _ = new MemoryStream();
            await Task.CompletedTask;
            await Task.CompletedTask;
            await Task.CompletedTask;
        }
    }
}
```

The report shows the using statement has 1 of 4 branches covered.

image

```xml
<?xml version="1.0" encoding="utf-8"?>
<coverage line-rate="1" branch-rate="0.5" version="1.9" timestamp="1623164584" lines-covered="7" lines-valid="7" branches-covered="3" branches-valid="6">
  <sources>
    <source>C:\</source>
  </sources>
  <packages>
    <package name="Lib" line-rate="1" branch-rate="0.5" complexity="6">
      <classes>
        <class name="Lib.Foo/&lt;Bar&gt;d__0" filename="\tmp\Lib\Foo.cs" line-rate="1" branch-rate="0.5" complexity="6">
          <methods>
            <method name="MoveNext" signature="()" line-rate="1" branch-rate="0.5" complexity="6">
              <lines>
                <line number="9" hits="1" branch="True" condition-coverage="100% (2/2)">
                  <conditions>
                    <condition number="16" type="jump" coverage="100%" />
                  </conditions>
                </line>
                <line number="10" hits="1" branch="False" />
                <line number="11" hits="1" branch="True" condition-coverage="25% (1/4)">
                  <conditions>
                    <condition number="135" type="switch" coverage="25%" />
                  </conditions>
                </line>
                <line number="12" hits="1" branch="False" />
                <line number="13" hits="1" branch="False" />
                <line number="14" hits="1" branch="False" />
                <line number="15" hits="1" branch="False" />
              </lines>
            </method>
          </methods>
          <lines>
            <line number="9" hits="1" branch="True" condition-coverage="100% (2/2)">
              <conditions>
                <condition number="16" type="jump" coverage="100%" />
              </conditions>
            </line>
            <line number="10" hits="1" branch="False" />
            <line number="11" hits="1" branch="True" condition-coverage="25% (1/4)">
              <conditions>
                <condition number="135" type="switch" coverage="25%" />
              </conditions>
            </line>
            <line number="12" hits="1" branch="False" />
            <line number="13" hits="1" branch="False" />
            <line number="14" hits="1" branch="False" />
            <line number="15" hits="1" branch="False" />
          </lines>
        </class>
      </classes>
    </package>
  </packages>
</coverage>
```

I used this test class.

```csharp
using System.Threading.Tasks;
using Xunit;

namespace Test
{
    public class FooTest
    {
        [Fact]
        public async Task Test()
        {
            await Lib.Foo.Bar();
        }
    }
}
```

### LINKED ISSUE #1177: Branch coverage Issue - async await
Hi again)
**Code:**

```
namespace CoverletAsync
{
    public class MyService
    {
        public async Task<byte[]> GetBytes(Func<Task<bool>> process, Func<Task<MemoryStream>> getStream, Func<ISession> getSession, Func<Stream, ISession, Task<byte[]>> doWithStream)
        {
            if (!await process())
            {
                return Array.Empty<byte>();
            }

            using var session = getSession();
            await using var stream = await getStream();
            return await doWithStream(stream, session);
        }

        public interface ISession : IDisposable
        {
        }
    }
}
```

**Test:**

```
namespace CoverletAsync.Tests
{
    public class Tests
    {
        [Test]
        public async Task SomeMethodEmpty()
        {
            await using var memoryStream = new MemoryStream();
            var actual = await new MyService().GetBytes(() => Task.FromResult(false), () => Task.FromResult(memoryStream), () => new TestSession(), (stream, session) => Task.FromResult(new byte[] { 0x01, 0x02 }));

            CollectionAssert.IsEmpty(actual);
        }

        [Test]
        public async Task SomeMethodNonEmpty()
        {
            await using var memoryStream = new MemoryStream();
            var actual = await new MyService().GetBytes(() => Task.FromResult(true), () => Task.FromResult(memoryStream), () => new TestSession(), (stream, session) => Task.FromResult(new byte[] { 0x01, 0x02 }));

            Assert.That(actual, Is.EqualTo(new byte[] { 0x01, 0x02 }));
        }

        private class TestSession : MyService.ISession
        {
            public void Dispose()
            {
            }
        }
    }
}
```

**Expected:**
Full branch coverage

**Actual:**
Partially covered (1 visits, 2 of 4 branches are covered)

**Output:**
image

**Report:**

```
<?xml version="1.0" encoding="utf-8"?>
<coverage line-rate="1" branch-rate="0.75" version="1.9" timestamp="1623166089" lines-covered="8" lines-valid="8" branches-covered="6" branches-valid="8">
  <sources>
    <source>F:\</source>
  </sources>
  <packages>
    <package name="CoverletAsync" line-rate="1" branch-rate="0.75" complexity="8">
      <classes>
        <class name="CoverletAsync.MyService/&lt;GetBytes&gt;d__0" filename="Work\experiments\CoverletAsync\CoverletAsync\MyService.cs" line-rate="1" branch-rate="0.75" complexity="8">
          <methods>
            <method name="MoveNext" signature="()" line-rate="1" branch-rate="0.75" complexity="8">
              <lines>
                <line number="10" hits="2" branch="True" condition-coverage="100% (2/2)">
                  <conditions>
                    <condition number="16" type="jump" coverage="100%" />
                  </conditions>
                </line>
                <line number="11" hits="2" branch="True" condition-coverage="100% (2/2)">
                  <conditions>
                    <condition number="146" type="jump" coverage="100%" />
                  </conditions>
                </line>
                <line number="12" hits="1" branch="False" />
                <line number="13" hits="1" branch="False" />
                <line number="16" hits="1" branch="True" condition-coverage="50% (2/4)">
                  <conditions>
                    <condition number="181" type="switch" coverage="50%" />
                  </conditions>
                </line>
                <line number="17" hits="1" branch="False" />
                <line number="18" hits="1" branch="False" />
                <line number="19" hits="2" branch="False" />
              </lines>
            </method>
          </methods>
          <lines>
            <line number="10" hits="2" branch="True" condition-coverage="100% (2/2)">
              <conditions>
                <condition number="16" type="jump" coverage="100%" />
              </conditions>
            </line>
            <line number="11" hits="2" branch="True" condition-coverage="100% (2/2)">
              <conditions>
                <condition number="146" type="jump" coverage="100%" />
              </conditions>
            </line>
            <line number="12" hits="1" branch="False" />
            <line number="13" hits="1" branch="False" />
            <line number="16" hits="1" branch="True" condition-coverage="50% (2/4)">
              <conditions>
                <condition number="181" type="switch" coverage="50%" />
              </conditions>
            </line>
            <line number="17" hits="1" branch="False" />
            <line number="18" hits="1" branch="False" />
            <line number="19" hits="2" branch="False" />
          </lines>
        </class>
      </classes>
    </package>
  </packages>
</coverage>
```

Run: dotnet test /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura

Reproduced in .NET Core 3.1, Coverlet 3.0.3 / Coverlet 3.0.4-preview.31.g4902c245c8

### PR DESCRIPTION
**Pull request by daveMueller**
closes #1177
closes #1176

### CODE REVIEW

File: src/coverlet.core/Symbols/CecilSymbolHelper.cs [hunk 1]
```diff
@@ -125,7 +125,7 @@ private static bool SkipMoveNextPrologueBranches(Instruction instruction)
                    that check state machine value to jump to correct state (for instance after a true async call)
                    Check if it's a Cond_Branch on state machine current value int num = <>1__state;
                    We are on branch OpCode so we need to go back by max 2 operation to reach ldloc.0 the load of "num"
```
Comment (strout) [inline src/coverlet.core/Symbols/CecilSymbolHelper.cs:127]:
```suggestion
                   We are on branch OpCode so we need to go back by max 3 operation to reach ldloc.0 the load of "num"
```

Comment (daveMueller) [reply src/coverlet.core/Symbols/CecilSymbolHelper.cs:127]: Totally missed that, thanks. :thumbsup:

**Comment by g0dm0d3** [comment]
Could someone do a code review, please ?

**Review by MarcoRossignoli — APPROVED** [review]
LGTM thanks!

**Comment by g0dm0d3** [comment]
Thanks, guys!

**Pull request merged**